layers/+distribution/spacemacs-base/funcs.el

```diff
@@ -402,6 +402,14 @@ removal."
       (buffer-disable-undo)
       (fundamental-mode))))
 
+;; our own implementation of kill-this-buffer from menu-bar.el
+(defun spacemacs/kill-this-buffer ()
+  "Kill the current buffer."
+  (interactive)
+  (if (window-minibuffer-p)
+    (abort-recursive-edit)
+    (kill-buffer (current-buffer))))
```

Comment:
Are you sure the `kill-buffer` sexp should be _inside_ the `if` ? If yes then it is not indented properly, if no you should use `when` instead of `if`.